modules/issue_tracker/php/uploadhelper.class.inc

```
@@ -0,0 +1,237 @@
+    {
+        $config = \NDB_Config::singleton();
+        $attachment_data_dir = trim(
+            rtrim(
+                $config->getSetting('IssueTrackerDataPath'),
+                '/'
+            )
+        );
+
+        if (is_writable($attachment_data_dir)) {
+            $fileToUpload->full_path = $attachment_data_dir .
+                                       '/attachments';
+            if (!is_dir($fileToUpload->full_path)) {
+                mkdir($fileToUpload->full_path, 0770, true);
+            }
+        } else {
+            throw new \LorisException(
+                'Issue_Tracker attachments directory not writable.'
+            );
+        }
+
+    }
+
+    /**
+     * This insert a record in the
+     * 'issues_attachments' table.
+     *
+     * @param object $fileToUpload The object containing
+     *                             the $_FILES and the $_POST values.
+     *
+     * @return void
+     */
+    function registerFile(Object &$fileToUpload) : void
+    {
+        $DB     = \NDB_Factory::singleton()->database();
+        $values = array(
+            'issueID'     => $this->_values['issueID'],
+            'file_hash'   => $fileToUpload->file_hash,
+            'date_added'  => date('Y-m-d h:i:s', time()),
+            'file_name'   => $fileToUpload->file,
+            'deleted'     => 0,
+            'user'        => $fileToUpload->inserted_by,
+            'description' => $fileToUpload->description,
+            'file_size'   => $fileToUpload->size,
+            'mime_type'   => $fileToUpload->mime_type
+        );
+        try {
+            $DB->insert('issues_attachments', $values);
+            // Move file from tmp to issue_tracker attachment directory.
+            $path = $fileToUpload->full_path . '/' . $fileToUpload->file_hash;
+            move_uploaded_file($_FILES['file']['tmp_name'], $path);
+        } catch (\DatabaseException $e) {
+            $this->endWithFailure();
+            error_log($e->getMessage());
```

It should probably not fail silently. It could throw the the Exception again.

---

@xlecours thanks.